materialize/resources/resource_connection.go

```diff
@@ -0,0 +1,838 @@
+			q.WriteString(fmt.Sprintf(`BROKER '%s',`, b.kafkaBroker))
+		}
+		if b.kafkaBrokers != nil {
+			if b.kafkaSSHTunnel != "" {
+				q.WriteString(`BROKERS (`)
+				for i, broker := range b.kafkaBrokers {
+					q.WriteString(fmt.Sprintf(`'%s' USING SSH TUNNEL %s`, broker, b.kafkaSSHTunnel))
+					if i < len(b.kafkaBrokers)-1 {
+						q.WriteString(`,`)
+					}
+				}
+				q.WriteString(`),`)
+			} else {
+				q.WriteString(fmt.Sprintf(`BROKERS (%s),`, strings.Join(b.kafkaBrokers, ",")))
+			}
+		}
+		if b.kafkaProgressTopic != "" {
+			q.WriteString(fmt.Sprintf(`PROGRESS TOPIC '%s',`, b.kafkaProgressTopic))
+		}
+		if b.kafkaSSLCa != "" {
+			q.WriteString(fmt.Sprintf(`SSL CERTIFICATE AUTHORITY SECRET %s,`, b.kafkaSSLCa))
+		}
+		if b.kafkaSSLCert != "" {
+			q.WriteString(fmt.Sprintf(`SSL CERTIFICATE SECRET %s,`, b.kafkaSSLCert))
+		}
+		if b.kafkaSSLKey != "" {
+			q.WriteString(fmt.Sprintf(`SSL KEY SECRET %s,`, b.kafkaSSLKey))
+		}
+		if b.kafkaSASLMechanisms != "" {
+			q.WriteString(fmt.Sprintf(`SASL MECHANISMS '%s',`, b.kafkaSASLMechanisms))
+		}
+		if b.kafkaSASLUsername != "" {
+			q.WriteString(fmt.Sprintf(`SASL USERNAME '%s',`, b.kafkaSASLUsername))
+		}
+		if b.kafkaSASLPassword != "" {
+			q.WriteString(fmt.Sprintf(`SASL PASSWORD SECRET %s`, b.kafkaSASLPassword))
+		}
+
+	}
+
+	if b.connectionType == "CONFLUENT SCHEMA REGISTRY" {
+		q.WriteString(fmt.Sprintf(`URL '%s'`, b.confluentSchemaRegistryUrl))
+		if b.confluentSchemaRegistryUsername != "" {
+			q.WriteString(fmt.Sprintf(`, USERNAME = '%s'`, b.confluentSchemaRegistryUsername))
+		}
+		if b.confluentSchemaRegistryPassword != "" {
+			q.WriteString(fmt.Sprintf(`, PASSWORD = SECRET %s`, b.confluentSchemaRegistryPassword))
+		}
+		if b.confluentSchemaRegistrySSLCa != "" {
+			q.WriteString(fmt.Sprintf(`, SSL CERTIFICATE AUTHORITY = %s`, b.confluentSchemaRegistrySSLCa))
+		}
+		if b.confluentSchemaRegistrySSLCert != "" {
+			q.WriteString(fmt.Sprintf(`, SSL CERTIFICATE = %s`, b.confluentSchemaRegistrySSLCert))
+		}
+		if b.confluentSchemaRegistrySSLKey != "" {
+			q.WriteString(fmt.Sprintf(`, SSL KEY = %s`, b.confluentSchemaRegistrySSLKey))
+		}
+		if b.confluentSchemaRegistryAWSPrivateLink != "" {
+			q.WriteString(fmt.Sprintf(`, AWS PRIVATELINK %s`, b.confluentSchemaRegistryAWSPrivateLink))
+		}
+		if b.confluentSchemaRegistrySSHTunnel != "" {
+			q.WriteString(fmt.Sprintf(`, SSH TUNNEL %s`, b.confluentSchemaRegistrySSHTunnel))
+		}
+	}
+
+	q.WriteString(`);`)
+	return q.String()
+}
+
+func (b *ConnectionBuilder) Read() string {
+	return fmt.Sprintf(`
+		SELECT
+			mz_connections.id,
+			mz_connections.name,
+			mz_connections.type,
+			mz_schemas.name AS schema_name
+		FROM mz_connections
+		JOIN mz_schemas
+			ON mz_connections.schema_id = mz_schemas.id
+		WHERE mz_connections.name = '%s'
+		AND mz_schemas.name = '%s';
+	`, b.connectionName, b.schemaName)
+}
+
+func (b *ConnectionBuilder) Rename(newConnectionName string) string {
+	q := strings.Builder{}
```

Comment:
Nit: These simple string parsing you don't have to use `strings.Builder{}` Could probably just be
```
return fmt.Sprintf(`ALTER CONNECTION %s.%s RENAME TO %s.%s;`, b.schemaName, b.connectionName, b.schemaName, newConnectionName)
```

Reply:
I like that! Just updated it!